src/wallet/rpcwallet.cpp

```diff
@@ -2918,7 +3056,8 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
             "z_getbalance \"address\" ( minconf inZat )\n"
             "\nReturns the balance of a taddr or zaddr belonging to the node's wallet.\n"
             "\nCAUTION: If the wallet has only an incoming viewing key for this address, then spends cannot be"
-            "\ndetected, and so the returned balance may be larger than the actual balance.\n"
+            "\ndetected, and so the returned balance may be larger than the actual balance."
+            "\nThe argument address may not be a unified address; please use z_getbalanceforaddress instead.\n"
```

Comment:
```suggestion
            "\nThe argument address may not be a Unified Address; please use z_getbalanceforaddress instead.\n"
```

Reply:
Open question: should this instead add together the balances of the constituent addresses when given a UA?

Reply:
It would be possible to implement this, but it would be a bit fiddly; we'd have to look for notes with the diversifier corresponding to that address. Given that UAs are all diversified, I think it'd be better to only give balances by account; if someone wants this functionality they can ask for it (or submit a PR for it) later.